src/librustc_codegen_utils/symbol_names.rs

```diff
@@ -398,12 +400,24 @@ impl ItemPathBuffer for SymbolPathBuffer {
     }
 }
 
+// Manual Debug implementation to omit non-Debug `tcx` field.
+impl fmt::Debug for SymbolPathBuffer<'_, '_> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.debug_struct("SymbolPathBuffer")
+            .field("result", &self.result)
```

Comment:
Do not omit the `tcx` field entirely, just give it some dummy value such as `_`.